internal/ingress/types.go

```diff
@@ -148,6 +148,8 @@ type Server struct {
 	SSLCiphers string `json:"sslCiphers,omitempty"`
 	// AuthTLSError contains the reason why the access to a server should be denied
 	AuthTLSError string `json:"authTLSError,omitempty"`
+	// AllCustomHTTPErrors contains an array of custom error codes for all locations
+	AllCustomHTTPErrors []int `json:allCustomHTTPErrors,omitempty`
```

Comment:
Is this really necessary? A `server` has locations you can access when rendering at https://github.com/kubernetes/ingress-nginx/pull/3344/files#diff-980db9e4b88704f12338bd074839f94eR821 and aggregate the custom HTTP codes per location. Alo IMO template.go is a better place for doing the aggregation rather than controller.go.

Reply:
@ElvinEfendi IIRC I set this up when I intended to do deduplication (for example - two ingresses for the same host but different paths can both intercept `404`, but only one `@custom_404` should be created). I looked into doing this directly in the gotemplate but it seemed tricky - I'll look into moving this into `template.go` as you suggest

Reply:
@ElvinEfendi as per your comment I removed the `AllCustomHTTPErrors` field and replaced it with a template function to do the collation of all error routes for a server block, and added deduplication (so both ingresses can specify 404 and only one `@custom_404` location is created, for example).

Let me know if this is more in line with what you were thinking!